sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/PreDefinedTagsResourceIdentifier.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// The identifier of a resource that is contained in a subscription.
+    /// </summary>
+    public class PreDefinedTagsResourceIdentifier : SubscriptionResourceIdentifier
```

Comment:
I don't think we want a new resourceid type, we just want to modify the parsing logic to handle /tagNames/{tagName] coming after a subscription segment.  Add new test cases to validate

Reply:
Done